app/models/user.rb

```diff
@@ -0,0 +1,19 @@
+class User < ActiveRecord::Base
+  has_secure_password
+  has_many :tweets
+
+  def slug
+    self.username.gsub(" ", "-").downcase
+  end
+
+  def self.find_by_slug(slug)
+    found = nil
+    User.all.each do |user|
+      if user.slug == slug
+        found = user
+      end
+    end
+    found
```

Comment:
Instead of redefining your own `find_by_slug` method, you can utilize the existing ActiveRecord `find_by` method.

```ruby
def self.find_by_slug(slug)
  User.find_by(slug: slug)
end
```
This returns an ActiveRecord::Collection object, allowing you to chain additional ActiveRecord methods.